File: src/main/java/io/openfuture/openmessanger/web/response/MessageResponse.java
package io.openfuture.openmessanger.web.response;

import java.time.LocalDateTime;

import io.openfuture.openmessanger.repository.entity.MessageContentType;

public record MessageResponse(
        int id,
        String sender,
        String recipient,
        String content,
        MessageContentType contentType,
        LocalDateTime receivedAt,
        LocalDateTime sentAt) {
}

File: src/main/kotlin/io/openfuture/openmessanger/web/response/MessageResponse.kt
package io.openfuture.openmessanger.web.response

import io.openfuture.openmessanger.repository.entity.MessageContentType
import java.time.LocalDateTime

data class MessageResponse(
    var id: Int,
    var sender: String,
    var recipient: String,
    var content: String,
    var contentType: MessageContentType,
    var receivedAt: LocalDateTime,
    var sentAt: LocalDateTime,
    var privateChatId: Int,
    var groupChatId: Int?,
    var attachments: List<Int> = emptyList()
)